server/routes/auth.py

```diff
@@ -102,76 +142,301 @@ def post(self):
+
+        data = request.get_json(silent=True) or {}
+        new_password = data.get("new_password")
+
+        if not new_password:
+            return {
+                "status": "error",
+                "message": "New password is required",
+                "data": None,
+            }, 400
+
+        if not is_valid_password(new_password):
+            return {
+                "status": "error",
+                "message": (
+                    "Password must have at least 8 characters, "
+                    "one uppercase letter, and one digit."
+                ),
+                "data": None,
+            }, 400
+
+        # Save new password
+        user.set_password(new_password)
+        db.session.commit()
+
+        return {
+            "status": "success",
+            "message": "Password reset successful",
+            "data": None,
+        }, 200
+
+
+# ------------------------------------------------------------
+# CHANGE PASSWORD
+# Logged-in user changes password (requires old + new)
+# ------------------------------------------------------------
+class ChangePasswordResource(Resource):
+    @jwt_required()
+    def post(self):
+        uid = int(get_jwt_identity())
+        user = User.query.get(uid)
+        if not user:
+            return {
+                "status": "error",
+                "message": "User not found",
+                "data": None,
+            }, 404
+
+        data = request.get_json(silent=True) or {}
+        current_password = data.get("current_password")
+        new_password = data.get("new_password")
+
+        if not current_password or not new_password:
+            return {
+                "status": "error",
+                "message": "Current and new password are required",
+                "data": None,
+            }, 400
+
+        if not user.check_password(current_password):
+            return {
+                "status": "error",
+                "message": "Current password is incorrect",
+                "data": None,
+            }, 401
+
+        if not is_valid_password(new_password):
+            return {
+                "status": "error",
+                "message": (
+                    "Password must have at least 8 characters, "
+                    "one uppercase letter, and one digit."
+                ),
+                "data": None,
+            }, 400
+
+        user.set_password(new_password)
+        db.session.commit()
+
+        return {
+            "status": "success",
+            "message": "Password changed successfully",
+            "data": None,
+        }, 200
 
 
-# Attach resource to /verify route
+# ------------------------------------------------------------
+# Register all routes
+# ------------------------------------------------------------
 api.add_resource(VerifyResource, "/verify")
 api.add_resource(LoginResource, "/login")
 api.add_resource(LogoutResource, "/logout")
 api.add_resource(RefreshResource, "/refresh")
```

Comment:
I know you worked on these as well. Please confirm they have contracts and examples as well.